File: Adorable/app/src/main/java/merryweather/com/adorable/di/component/FragmentComponent.java
package merryweather.com.adorable.di.component;

import android.content.Context;

import dagger.Component;
import merryweather.com.adorable.di.modules.FragmentModule;
import merryweather.com.adorable.di.scopes.PerFragment;


@PerFragment
@Component(dependencies = {ActivityComponent.class}, modules = FragmentModule.class)
public interface FragmentComponent {

    Context context();


}

File: Nista/app/src/main/java/com/example/nista/di/component/FragmentComponent.kt
package com.example.nista.di.component

import android.content.Context
import com.example.nista.di.component.ActivityComponent
import com.example.nista.di.modules.FragmentModule
import com.example.nista.di.scopes.PerFragment
import com.leroymerlin.cds.di.modules.ViewModelModule

import dagger.Component


@PerFragment
@Component(dependencies = [ActivityComponent::class], modules = arrayOf(FragmentModule::class, ViewModelModule::class))
interface FragmentComponent {

    fun context() : Context



}